p2panda-net/src/engine/engine.rs

```diff
@@ -410,7 +566,16 @@ impl EngineActor {
         );
 
         // Register earmarked topics from other peers
-        self.peers.on_announcement(topics, delivered_from);
+        self.peers.on_announcement(topics.clone(), delivered_from);
+
+        // Inform the connection manager about the peer topics
+        if let Some(tx) = &self.connection_actor_tx {
+            tx.send(ToConnectionActor::UpdatePeerTopics {
+                peer: delivered_from,
+                topics,
+            })
+            .await?;
+        }
```

Comment:
@adzialocha do you agree that this is the correct place to notify the connection manager of a new peer and/or topics? This is the point where the connection manager discovers peers, internally it decides whether to initiate sync based on 1) is an outbound sync session in progress already for this peer+topic? 2) did we already complete a sync session with this peer+topic? If the answer to either of those is "yes" then we decide not to initiate sync.

Reply:
Yes! Looks good.

This is the place where "discovery" finally brought the peer and its topics together.

Reply:
Great! Just wanted to check :smile: